src/EVM/SymExec.hs

```diff
@@ -321,13 +321,29 @@ interpret fetcher maxIter askSmtIters heuristic vm =
         conf <- readConfig
         (r, vm') <- liftIO $ stToIO $ runStateT (exec conf) vm
         interpret fetcher maxIter askSmtIters heuristic vm' (k r)
+      Stepper.ForkMany (PleaseRunAll expr vals continue) -> do
+        when (length vals < 2) $ internalError "PleaseRunAll requires at least 2 branches"
+        frozen <- liftIO $ stToIO $ freezeVM vm
+        let newDepth = vm.exploreDepth+1
+        threads <- withRunInIO $ \runInIO -> mapConcurrently (runInIO . runOne frozen newDepth) vals
+        pure $ goITE (zip vals threads)
```

Comment:
Is `threads` the best name? The type of the elements is `Expr End`, right?
So aren't these really `branches`?

Also, can you comment for my understanding what is going to happen in Haskell runtime?
Is it going to spawn all threads immediately, or is there some thread pool with a limit on the number of threads that can run concurrently?

Reply:
Yes, true. They are branches. I'll rename.

It will spawn them all immediately. Unfortunately, there is no thread pool. Yes, this can actually lead to issues, but this has been known already. I mentioned this to Lexi, but they were quite sure it's fine. And honestly, this thread issue has not caused any problems (they could, eating up memory), so Lexi was right :)